x-pack/plugins/screenshotting/server/screenshots/index.test.ts

```diff
@@ -82,13 +83,18 @@ describe('Screenshot Observable Pipeline', () => {
 
     screenshots = new Screenshots(driverFactory, logger, packageInfo, http, config, cloud);
 
-    jest.spyOn(Layouts, 'createLayout').mockReturnValue(layout);
-
+    // Using this patch instead of using `jest.spyOn`. This way we avoid calling
+    // `jest.restoraAllMocks()` which removes implementations from other mocks not
+    // explicit in this test (like apm mock object)
+    // @ts-expect-error
+    Layouts.createLayout = () => layout;
```

Comment:
Note for reviewers: although Jest [documentation](https://jestjs.io/docs/jest-object#jestrestoreallmocks) say `.restoreAllMocks` only works for mocks created by `spyOn` I found that methods from the mock agent from `@kbn/test` (./packages/kbn-test/src/jest/mocks/apm_agent_mock.ts) had their implementation reset the the basic `jest.fn()`. Hence `agent.startTransaction(...)` was returning `undefined` instead of a transaction.

I preferred to use this patching methods and not `jest.mock` since I find it more straightforward